distributor/distributor.go

```diff
@@ -320,6 +338,19 @@ func (d *Distributor) Push(ctx context.Context, req *remote.WriteRequest) (*cort
 	return &cortex.WriteResponse{}, nil
 }
 
+func (d *Distributor) getOrCreateIngestLimiter(userID string) *rate.Limiter {
+	d.ingestLimitersMtx.Lock()
```

Comment:
I'd have probably gone with a RWMutex, I expecting each ingester to be able to handle ~1million samples/s, or 10k http requests a second.  Right now they're only doing ~100 req/s, so its not a big deal.  We can change later.